utils/line-directive-file

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python
+# line-directive-windows.py - Error message line transformer ----*- python -*-
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See http://swift.org/LICENSE.txt for license information
+# See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+#
+# ----------------------------------------------------------------------------
+#
+# Converts line numbers in error messages according to "line directive"
+# comments.
+#
+# ----------------------------------------------------------------------------
+
+import sys
+import os
+import subprocess
+import platform
+
+def run():
+    # On Windows, the command line doesn't let us pass long files. This means
+    # that we need to have a way to bypass the command line to pass files to
+    # line-directive. An alternative is reading a file written by CMake and
+    # forwarding the arguments via Python.
+    with open(sys.argv[1], 'r') as files:
```

Comment:
Adding a:

    if sys.argv[0][0] == '@':

means that we can integrate this into the original tool and makes it behave more like other tools.